src/cli_plugin/install/kibana.js

```diff
@@ -0,0 +1,56 @@
+import _ from 'lodash';
+import fromRoot from '../../utils/fromRoot';
+import KbnServer from '../../server/KbnServer';
+import readYamlConfig from '../../cli/serve/read_yaml_config';
+import { statSync } from 'fs';
+
+export function existingInstall(settings, logger) {
+  try {
+    statSync(settings.plugins[0].path);
+
+    logger.error(`Plugin ${settings.plugins[0].name} already exists, please remove before installing a new version`);
+    process.exit(70); // eslint-disable-line no-process-exit
+  } catch (e) {
+    if (e.code !== 'ENOENT') throw e;
+  }
+}
+
+export async function rebuildCache(settings, logger) {
+  logger.log('Optimizing and caching browser bundles...');
+  const serverConfig = _.merge(
+    readYamlConfig(settings.config),
+    {
+      env: 'production',
+      logging: {
+        silent: settings.silent,
+        quiet: !settings.silent,
+        verbose: false
+      },
+      optimize: {
+        useBundleCache: false
+      },
+      server: {
+        autoListen: false
+      },
+      plugins: {
+        initialize: false,
+        scanDirs: [settings.pluginDir, fromRoot('src/plugins')]
+      }
+    }
+  );
+
+  const kbnServer = new KbnServer(serverConfig);
+  await kbnServer.ready();
+  await kbnServer.close();
```

Comment:
This seems like a pretty heavy-handed approach to managing the cache.  Is there a way we can expose the optimize capabilities of kibana without standing up an entire server?